olp-cpp-sdk-core/include/olp/core/network2/NetworkResponse.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include <olp/core/CoreApi.h>
+
+namespace olp {
+namespace network2 {
+/**
+ * @brief The NetworkResponse class represents network response abstraction for
+ * a HTTP request
+ */
+struct CORE_API NetworkResponse {
+ public:
+  /**
+   * @brief NetworkResponse default constructor
+   */
+  NetworkResponse();
+
+  bool is_cancelled;
```

Comment:
please align with code styles.
What about all other fields: id, payload_size, payload_offset, max_age, expires?
I know they are part of the headers, but still useful for the user to have them part of the response.